paasta_tools/cli/cmds/spark_run.py

```diff
@@ -332,13 +332,14 @@ def get_docker_run_cmd(container_name, volumes, env, docker_img, docker_cmd, nvi
     return cmd
 
 
-def get_smart_paasta_instance_name(instance):
+def get_smart_paasta_instance_name(instance, cmd):
     if os.environ.get("TRON_JOB_NAMESPACE"):
         tron_job = os.environ.get("TRON_JOB_NAME")
         tron_action = os.environ.get("TRON_ACTION")
         return f"{tron_job}.{tron_action}"
     else:
-        return f"{instance}_{get_username()}"
+        how_submitted = cmd.split(" ")[0]
+        return f"{instance}_{get_username()}_{how_submitted}"
```

Comment:
The one case this doesn't handle is the mrjob spark runner.  See for instance https://sourcegraph.yelpcorp.com/sysgit/yelpsoa-configs@ad7081eeafa1119137db6def88b8097df87e9626/-/blob/ad_creative/tron-norcal-prod.yaml#L90 -- the command is just `python -m ...`

I think the move for that case is just to look at the `--mrjob` arg they're passing.